### PR TITLE
Remove net451 as a cross-compile target

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/EFCore.Design/Design/Internal/ForwardingProxy.cs [hunk 1]
```diff
@@ -18,10 +18,12 @@ public static class ForwardingProxy
...
             where T : class
         {
-#if NET451
+#if NET46
             return target as T ?? new ForwardingProxy<T>(target).GetTransparentProxy();
-#else
+#elif NETSTANDARD1_3
             return (T)target;
+#else
+#error target frameworks need to be updated.
```
Comment (bricelam) [inline src/EFCore.Design/Design/Internal/ForwardingProxy.cs]: Not a fan, but I can see the value...

File: src/EFCore.Specification.Tests/TestModels/ConcurrencyModel/Team.cs [hunk 2]
```diff
@@ -10,10 +10,12 @@ namespace Microsoft.EntityFrameworkCore.Specification.Tests.TestModels.Concurren
 {
     public class Team
     {
-#if NET452
+#if NET46
         private readonly ObservableCollection<Driver> _drivers = new ObservableCollectionListSource<Driver>();
```
Comment (pranavkm) [inline src/EFCore.Specification.Tests/TestModels/ConcurrencyModel/Team.cs]: Btw, I wasn't sure if you still need the observable types. Are they replaceable by the ones in `NETStandard1.3`?

Comment (bricelam) [inline src/EFCore.Specification.Tests/TestModels/ConcurrencyModel/Team.cs]: @ajcvickers?

Comment (pranavkm) [inline src/EFCore.Specification.Tests/TestModels/ConcurrencyModel/Team.cs]: I'll merge the PR in to unblock the rest of my work. You can remove this in a follow up if it is indeed removable.

**Pull request closed without merging**